// components/buttons/styled-buttons.tsx
import styled, { css } from 'styled-components';

const Button = styled.button`
	background: transparent;
	padding: 0 14px;
	min-width: 80px;
	height: 32px;
	font-family: 'Roboto';
	font-style: normal;
	font-weight: 500;
	line-height: normal;
	font-size: 13px;
	border-radius: 2px;
	outline: none;
	border-width: 0px;
	transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
		border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
	:focus {
		box-shadow: 0px 0px 2px 2px #9ac3ff;
		outline: none;
	}
`;

const Primary = styled(Button)`
	background: ${props => props.theme.$blue};
	color: ${props => props.theme.$white};
	border-width: 0px;
	:hover {
		background: #1d4db7;
	}
	:active {
		background: #103fa5;
	}
	:disabled {
		background: #8fadf1;
	}
`;

const Secondary = styled(Button)`
	background: #ffffff;
	border: 1px solid #b7d4ff;
	color: ${props => props.theme.$blue};
	:hover {
		color: #2353bd;
		border-color: #3564cf;
	}
	:focus {
		color: #3564cf;
		border-color: #3564cf;
	}
	:active {
		color: #103fa5;
		border-color: #103fa5;
	}
	:disabled {
		color: #afb6c1;
		padding-left: 13px;
		padding-right: 13px;
		border: 1px solid rgba(208, 215, 226, 0.8);
		background: #f3f6f8;
	}
`;

const Square = styled(Secondary)`
	min-width: 32px;
	padding: 0;
`;

const rounded = css`
	font-weight: bold;
	font-size: 15px;
	font-family: Lato;
	border-radius: 50px;
`;

const RoundedPrimary = styled(Primary)`
	${rounded}
`;

const RoundedSecondary = styled(Secondary)`
	${rounded}
`;

export const StyledButtons = {
	Button: {
		Base: Button,
		Primary,
		Secondary,
		Rounded: {
			Primary: RoundedPrimary,
			Secondary: RoundedSecondary,
		},
		Square,
	},
};
